src/libraries/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -717,6 +728,12 @@
   <data name="InvalidIndentSize" xml:space="preserve">
     <value>Indentation size must be between {0} and {1}.</value>
   </data>
+  <data name="PipeWriterCanceled" xml:space="preserve">
+    <value>PipeWriter.FlushAsync was canceled.</value>
```

Comment:
This feels like a message that should belong to the pipelines layer.

Reply:
But it's specific to Json's usage. Pipes doesn't throw when flush is canceled or the Pipe is completed. If just returns a `FlushResult` with the appropriate flags set and it's up to the user to decide what to do. Since we're in the middle of writing Json content it seems best to throw, as that's the only indicator to users we can give in this method, that we weren't able to write everything.

Reply:
Yeah, I realized that after looking where it's being used. Sounds good then 👍